packages/wallets/nightly/src/adapter.ts

```diff
@@ -0,0 +1,150 @@
+
+    private _connecting: boolean;
+    private _connected: boolean;
+    private _publicKey: PublicKey | null;
+    private _wallet: SolanaNightly | null;
+    private _readyState: WalletReadyState =
+        typeof window === 'undefined' || typeof document === 'undefined'
+            ? WalletReadyState.Unsupported
+            : WalletReadyState.NotDetected;
+
+    constructor() {
+        super();
+        this._connecting = false;
+        this._connected = false;
+        this._publicKey = null;
+        this._wallet = null;
+
+        if (this._readyState !== WalletReadyState.Unsupported) {
+            scopePollingDetectionStrategy(() => {
+                if (window?.nightly?.solana) {
+                    this._readyState = WalletReadyState.Installed;
+                    this.emit('readyStateChange', this._readyState);
+                    return true;
+                }
+                return false;
+            });
+        }
+    }
+
+    get connected(): boolean {
+        return this._connected;
+    }
+
+    get connecting(): boolean {
+        return this._connecting;
+    }
+
+    get readyState(): WalletReadyState {
+        return this._readyState;
+    }
+
+    public async signAllTransactions(transactions: Transaction[]): Promise<Transaction[]> {
+        try {
+            if (!this._wallet) {
+                throw new WalletNotConnectedError();
+            }
+
+            try {
+                return await this._wallet.signAllTransactions(transactions);
+            } catch (error: any) {
+                throw new WalletSignTransactionError(error?.message, error);
+            }
+        } catch (error: any) {
+            this.emit('error', error);
+            throw error;
+        }
+    }
+
+    get publicKey() {
+        return this._publicKey;
+    }
+
+    async signTransaction(transaction: Transaction) {
+        try {
+            if (!this._wallet) {
+                throw new WalletNotConnectedError();
+            }
+
+            try {
+                return await this._wallet.signTransaction(transaction);
+            } catch (error: any) {
+                throw new WalletSignTransactionError(error?.message, error);
+            }
+        } catch (error: any) {
+            this.emit('error', error);
+            throw error;
+        }
+    }
+
+    async connect(onDisconnect?: () => void) {
+        try {
+            if (this.connected || this.connecting) return;
+            if (this._readyState !== WalletReadyState.Installed) throw new WalletNotReadyError();
+
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            const wallet = window!.nightly!.solana!;
+
+            try {
+                this._connecting = true;
+                await wallet.connect(onDisconnect);
```

Comment:
Similarly, if you need to acquire a reference to `onDisconnect`, this has to be implemented here (by adding a method to handle disconnection, as seen in other adapters).

Reply:
removed